File: Cs/Solution/FindPoisonedDuration.cs

using System;

namespace Solution
{

	public partial class Solution
	{
		public int FindPoisonedDuration(int[] timeSeries, int duration)
		{
			int total = duration;
			for (int i = 1; i < timeSeries.Length; i++)
			{
				total += Math.Min(duration, timeSeries[i] - timeSeries[i - 1]);
			}
			return total;
		}
	}
}

File: Rust/src/solution/find_poisoned_duration.rs
use super::base::Solution;

impl Solution {
    pub fn find_poisoned_duration(time_series: Vec<i32>, duration: i32) -> i32 {
	let mut total = duration;
	for i in 1..time_series.len() {
	    total += std::cmp::min(time_series[i] - time_series[i - 1], duration);
	}
	total
    }
}
